src/onboarding/steps.test.ts

```diff
@@ -0,0 +1,237 @@
+import { BIOMETRY_TYPE } from 'react-native-keychain'
+import { initializeAccount } from 'src/account/actions'
+import { setHasSeenVerificationNux } from 'src/identity/actions'
+import {
+  navigate,
+  navigateClearingStack,
+  navigateHome,
+  popToScreen,
+} from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import {
+  END_OF_ONBOARDING_SCREEN,
+  FIRST_ONBOARDING_SCREEN,
+  getOnboardingStepValues,
+  goToNextOnboardingScreen,
+  _getScreenInfo,
+} from 'src/onboarding/steps'
+import { store } from 'src/redux/store'
+import { mockOnboardingProps } from 'test/values'
+import { mocked } from 'ts-jest/utils'
+
+jest.mock('src/redux/store', () => ({ store: { dispatch: jest.fn() } }))
+
+const mockStore = mocked(store)
+
+describe('onboarding steps', () => {
+  describe('_getScreenInfo and getOnboardingStepValues', () => {
+    it('All paths from FIRST_ONBOARDING_SCREEN lead to END_OF_ONBOARDING_SCREEN and have incrementing steps', () => {
+      const options = {
+        recoveringFromStoreWipe: [true, false],
+        choseToRestoreAccount: [true, false],
+        supportedBiometryType: [null, BIOMETRY_TYPE.FACE_ID],
+        skipVerification: [true, false],
+        numberAlreadyVerifiedCentrally: [true, false],
+      }
+
+      options.recoveringFromStoreWipe.forEach((recoveringFromStoreWipe) => {
+        options.choseToRestoreAccount.forEach((choseToRestoreAccount) => {
+          options.supportedBiometryType.forEach((supportedBiometryType) => {
+            options.skipVerification.forEach((skipVerification) => {
+              options.numberAlreadyVerifiedCentrally.forEach((numberAlreadyVerifiedCentrally) => {
```

Comment:
i think this could be a little cleaner if it were refactored as follows:
- take "options" out of the test body
- prep a list of cases to check by iterating over the options like you do here, and just pushing to an array of cases to check
- pass that list of cases to check into an it.each

you could also use [this SO post](https://stackoverflow.com/questions/12303989/cartesian-product-of-multiple-arrays-in-javascript) to prep the cases to check with a 1-line helper, but that's probably overkill

if nothing else, it would be nice to see this test written without all the nested indentation